cylc/flow/task_pool.py

```diff
@@ -724,6 +724,18 @@ def set_do_reload(self, config):
         self.max_num_active_cycle_points = (
             self.config.get_max_num_active_cycle_points())
 
+        # find any old tasks that have been removed from the suite
+        old_task_name_list = self.task_name_list
+        self.task_name_list = self.config.get_task_name_list()
+        for name in old_task_name_list:
+            if name not in self.task_name_list:
+                self.orphans.append(name)
+        for name in self.task_name_list:
+            if name in self.orphans:
+                self.orphans.remove(name)
```

Comment:
Set operations could be used here! However this will do (as they're only one off)